### ISSUE TITLE
[LifetimeSafety] False-negative with unique_ptr::reset()

### ISSUE THREAD

**Issue by usx95** [llvm/llvm-project#184630]
```cpp
#include <iostream>
#include <memory>
#include <string>
#include <vector>

int main() {
    int* ptr;
    auto p = std::make_unique<int>(1);
    ptr = p.get();
    p.reset();
    std::cout << *ptr;
}
```
https://godbolt.org/z/rcTdbjT1M

It should be possible to model this as an "invalidation" if `unique_ptr::reset` is considered to be invalidating.
